llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -87,12 +87,43 @@ bool llvm::canDelayMemOp(MachineInstr &MemI, MachineInstr &Dest,
   auto InstrRange = make_range(MII, MIE);
   bool SawStore = MemI.mayStore();
   auto UnsafeToMovePast = [&](const MachineInstr &MI) {
-    return (isUseOf(MI, MemI) && !isTriviallyDead(MI, MRI)) ||
+    bool IsCopyBetween = false;
+    // We can delay an instruction after a copy, if the copy just
+    // connects MemI and Dest. After combining, this copy will be dead.
+    if (MI.isCopy() && MRI.hasOneNonDBGUse(MI.getOperand(1).getReg()) &&
```

Comment:
Should we also look at bitcasts here as well?

Reply:
Hi @gbossu, basically I saw copys for loads and bitcasts+copy for stores, and we are using this function to move loads down. I think we can keep bitcast out for now and handle it just in case it is really necessary.